.github/workflows/hlint.yml

```diff
@@ -0,0 +1,24 @@
+name: lint
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - 'releases/*'
+
+jobs:
+  hlint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: 'Set up HLint'
+      uses: rwe/actions-hlint-setup@v1
+      with:
+        version: '3.1.6'
+
+    - name: 'Run HLint'
+      uses: rwe/actions-hlint-run@v2
+      with:
+        path: '["src/", "test/"]'
+        fail-on: warning
```

Comment:
I haven't used `rwe/actions-hlint-setup` and `rwe/actions-hlint-run` actions before.

I would prefer to use official Haskell actions under the `haskell` github org:

* https://github.com/haskell/actions

But even then, I'm not sure what is the difference between those two actions and a simple job that just downloads the `hlint` executable and runs it. I used to configure `hlint` before with this simple setup:

* https://github.com/kowainik/stan/blob/dcbc8682a964ae6f642ae37adbd46f6545d39aa6/.github/workflows/ci.yml#L102-L117

Reply:
IIUC:
- `rwe/actions-hlint-setup` deals with the subtleties of the underlying OS (ie. download le `exe` on windows, and on)
- `rwe/actions-hlint-run` only run hlint on the new/changed files

I update the PR according to your feedback, I'm open to new comments/requests.